tests/test_app.py

```diff
@@ -220,13 +228,179 @@ def test_xarray_part(
     xarray_query_params,
     arctic_bounds: Tuple[float, float, float, float],
 ) -> None:
-    """Test /part endpoint for xarray backend"""
+    """Test /bbox endpoint for xarray backend"""
     response = app.get(
         f"/bbox/{','.join(str(coord) for coord in arctic_bounds)}.tif",
         params={
             **xarray_query_params,
-            "format": "tif",
         },
     )
     assert response.status_code == 200
     assert response.headers["content-type"] == "image/tiff; application=geotiff"
+
+    size = (10, 10)
+    with pytest.warns():
+        response = app.get(
+            f"/bbox/{','.join(str(coord) for coord in arctic_bounds)}/{'x'.join(str(x) for x in size)}.png",
+            params={
+                **xarray_query_params,
+            },
+        )
+    assert response.status_code == 200
+    assert response.headers["content-type"] == "image/png"
+
+    image_data = io.BytesIO(response.content)
+    Image.open(image_data)
+
+    # Check dimensions
+    # assert image.size == size, f"Expected image size {size}, but got {image.size}"
+
+
+def test_timeseries_statistics(
+    app: TestClient,
+    mocker,
+    mock_cmr_get_assets,
+    xarray_query_params,
+    arctic_geojson,
+) -> None:
+    """Test /timeseries/statistics endpoint
+
+    Since the /timeseries/statistics endpoint sends more requests to internal endpoints
+    we need to catch those requests and mock a response since we can't forward them to
+    the test client.
+    """
+    arctic_stats = deepcopy(arctic_geojson)
+    arctic_stats["properties"]["statistics"] = {
+        "sea_ice_fraction": {
+            "min": 0.0,
+            "max": 1.0,
+            "mean": 0.3,
+            "count": 4493.0,
+            "sum": 1463.7,
+            "std": 0.3,
+            "median": 0.0,
+            "majority": 0.0,
+            "minority": 0.28,
+            "unique": 87.0,
+            "histogram": [
+                [2322, 38, 56, 300, 536, 426, 427, 388],
+                [0.0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1.0],
+            ],
+            "valid_percent": 93.72,
+            "masked_pixels": 301.0,
+            "valid_pixels": 4493.0,
+            "percentile_2": 0.0,
+            "percentile_98": 0.99,
+        }
+    }
+
+    async def mock_timestep_request(url: str, **kwargs) -> Response:
```

Comment:
Note: maybe over complex but there might be way to create a proper app running in threads using https://github.com/banesullivan/server-thread (https://github.com/developmentseed/rio-viz/blob/776fb29e5ce2647cdb39d962e9bb0de833f74feb/rio_viz/app.py#L799-L818)

Reply:
In this case I am comfortable with relying on the mocked PNG responses since we have already have tests for the `/bbox` endpoint.